tools/test-backend

```diff
@@ -123,7 +123,7 @@ def block_internet():
                         "https://zulip.readthedocs.io/en/latest/testing.html#internet-access-inside-test-suits")
 
     httplib2.Http.request = internet_guard
-    requests.request = internet_guard
+    requests.request = lambda method, url, **kwargs: internet_guard(method, url, **kwargs)
```

Comment:
This does seem to be the case. I've just tried to narrow down the issue to:
```
import requests

from typing import Any

def internet_guard(*args, **kwargs):
    # type: (*Any, **Any) -> None
    raise Exception()

requests.request = internet_guard
```
This does cause a problem, since `mypy` or `mypy -2` is fine, but with --strict-optional, I get:
```
11: error: Incompatible types in assignment (expression has type Callable[[VarArg(Any), KwArg(Any)], None], variable has type Callable[[str, str, KwArg(Any)], Response])
```
Does that look like a reasonable test-case?

(In addition, if I include httplib2 with the above, I need --ignore-missing-imports (as per run-mypy), since httplib2 doesn't have a typeshed (cf requests). So the former might also have complained!)